cmd/resolutioncli/main.go

```diff
@@ -0,0 +1,184 @@
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	catalogd "github.com/operator-framework/catalogd/api/core/v1alpha1"
+
+	operatorsv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/bundlesanddependencies"
+	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/crdconstraints"
+	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/entity"
+	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/olm"
+)
+
+const pocMessage = `This command is a proof of concept for off-cluster resolution and is not intended for production use!
+
+Please provide your feedback and ideas via https://github.com/operator-framework/operator-controller/discussions/262`
+
+const (
+	flagNamePackageName    = "package-name"
+	flagNamePackageVersion = "package-version"
+	flagNamePackageChannel = "package-channel"
+	flagNameIndexRef       = "index-ref"
+	flagNameInputDir       = "input-dir"
+)
+
+var (
+	scheme = runtime.NewScheme()
+
+	codecs = serializer.NewCodecFactory(scheme)
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(operatorsv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(rukpakv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(catalogd.AddToScheme(scheme))
+}
+
+func main() {
+	fmt.Fprintf(os.Stderr, "\033[0;31m%s\033[0m\n", pocMessage)
+
+	ctx := context.Background()
+
+	var packageName string
+	var packageVersion string
+	var packageChannel string
+	var indexRef string
+	var inputDir string
+	flag.StringVar(&packageName, flagNamePackageName, "", "Name of the package to resolve")
+	flag.StringVar(&packageVersion, flagNamePackageVersion, "", "Version of the package")
+	flag.StringVar(&packageChannel, flagNamePackageChannel, "", "Channel of the package")
+	// TODO: Consider adding support of multiple refs
+	flag.StringVar(&indexRef, flagNameIndexRef, "", "Index reference (FBC image or dir)")
+	flag.StringVar(&inputDir, flagNameInputDir, "", "Directory containing Kubernetes manifests (such as Operator) to be used as an input for resolution")
+	flag.Parse()
+
+	if err := validateFlags(packageName, indexRef); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	err := run(ctx, packageName, packageVersion, packageChannel, indexRef, inputDir)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func validateFlags(packageName, indexRef string) error {
+	if packageName == "" {
+		return fmt.Errorf("missing required -%s flag", flagNamePackageName)
+	}
+
+	if indexRef == "" {
+		return fmt.Errorf("missing required -%s flag", flagNameIndexRef)
+	}
+
+	return nil
+}
```

Comment:
Nit: If these flags are required would it make more sense to make them positional arguments? IMO flags should be reserved for configurable options. 

I noticed we aren't using `spf13/cobra` for building the CLI commands, but it has some nice support for setting required number of arguments that we could use here and automatically generates a pretty nice help message that could be used to show the required positional arguments.

Reply:
I'm a fan of explicit arguments; it can get confusing with positional unless there's an explicit logic (e.g. `kubectl`'s _verb_ _type_ [_name_])

Reply:
Package name probably makes sense as a positional argument and everything else as flags. But I suggest that we leave it as is for now and come back to discuss details of the interface once start work on finalising this CLI.

We are not using cobra right now, but I think we should use it in the production-ready version.